rust-symcrypt/src/ecc/mod.rs

```diff
@@ -440,14 +452,30 @@ fn internal_new(curve: CurveType) -> Result<InnerEcCurve, SymCryptError> {
 }
 
 impl InnerEcCurve {
-    // new() returns a EcCurve associated with the provided CurveType.
-    pub(crate) fn new(curve: CurveType) -> &'static Self {
-        let ec_curve: &'static InnerEcCurve = match curve {
-            CurveType::NistP256 => &*NIST_P256,
-            CurveType::NistP384 => &*NIST_P384,
-            CurveType::Curve25519 => &*CURVE_25519,
-        };
-        ec_curve
+    // new() returns an `InnerEcCurve` associated with the provided CurveType.
+    pub(crate) fn new(curve: CurveType) -> Result<&'static Self, SymCryptError> {
+        // Match the provided curve type and access the corresponding static `Result`.
+        match curve {
+            // Returning the lazy_static reference to the curve.
+            // Casting back our own MemoryAllocation error instead of getting the error from the static reference.
+            // This is done to avoid the need to clone the error.
```

Comment:
Since this function also now returns a `Result<&'static Self, SymCryptError>`, can you not just directly return the static ref the same way you did before?

Reply:
It has to do with the change in return types, and the fact that we are not using `unwrap()` anymore. We can't directly return back what we get from the `lazy_static` because the initialization of a `lazy_static` reference should not fail. Previously we were `unwrap()`ing to enforce that it wouldn't fail but now that we've added an Error variant `Result<InnerEcCurve, SymCryptError>` we have to handle with this match statement. I will change the code to be bit more concise though.

Reply:
changed to `CurveType::NistP256 => NIST_P256.as_ref().map_err(|_| SymCryptError::MemoryAllocationFailure),` because its less verbose